assets/settings.lagoon.php

```diff
@@ -31,15 +31,48 @@
 
 // Lagoon database connection.
 if (getenv('LAGOON')) {
-  $databases['default']['default'] = [
-    'driver' => 'mysql',
-    'database' => getenv('MARIADB_DATABASE') ?: 'drupal',
-    'username' => getenv('MARIADB_USERNAME') ?: 'drupal',
-    'password' => getenv('MARIADB_PASSWORD') ?: 'drupal',
-    'host' => getenv('MARIADB_HOST') ?: 'mariadb',
-    'port' => getenv('MARIADB_PORT') ?: 3306,
-    'prefix' => '',
-  ];
+  $dbtype = getenv("DB_TYPE") ?: (!empty(getenv('POSTGRES_HOST')) ? 'pgsql' : 'mysql');
+
+  switch ($dbtype) {
+    case ('pgsql'):
+      $databases['default']['default'] = [
+        'driver' => 'pgsql',
+        'database' => getenv('POSTGRES_DATABASE') ?: 'drupal',
+        'username' => getenv('POSTGRES_USERNAME') ?: 'drupal',
+        'password' => getenv('POSTGRES_PASSWORD') ?: 'drupal',
+        'host' => getenv('POSTGRES_HOST') ?: 'postgresql',
+        'port' => $dbtype == 'pgsql' ? 5432 : (getenv('MARIADB_PORT') ?: 3306),
```

Comment:
the logic here should more closely follow the others - we wouldn't be in this case if dbtype != 'pgsql'?
```suggestion
        'port' => getenv('POSTGRES_PORT') ?: 5432,
```